### PR TITLE
discordstatus command

### PR DESCRIPTION
**Pull request by BenSegal855**
**Describe this PR and why it should be merged:**
This PR adds the discordstatus command (aliases: ds, discstatus, isdiscordbroke). The command pulls information from https://srhpyqt94yxb.statuspage.io/api/v2/summary.json and formats it into a nice neat embed for the users viewing pleasure. It also adds node-fetch as a dependency to do this.
**Status**
- [x] This PR has been tested and complies with this project's ESLint rules
- [x] This PR adds/modifes a command

**Semantic Versioning**
- [ ] This PR features bug fixes, or only style changes/refactorings, or no code changes
- [x] This PR adds features for users
- [ ] This PR switches this project to a new a new Discord API library/new Discord.js framework

### CODE REVIEW

File: src/languages/en-US.ts [hunk 1]
```diff
@@ -316,6 +316,9 @@ export default class extends Language {
 			FOOTER: this.randomDftba,
 			TITLE: 'Statistics'
 		},
+		COMMAND_DISCORD_STAUTS_DECRIPTION: 'See the current status of Discord.',
+		COMMAND_DISCORD_STAUTS_EMBED_DECRIPTION: (incident): string => `[Discord Status](https://discordstatus.com/)\n**Current Incident:**\n${incident}`,
+		COMMAND_DISCORD_STATUS_EMBED_FOOTER: (time): string => `Last changed: ${time} • ${this.randomDftba}`,
```
Comment (jwford) [inline src/languages/en-US.ts]: Embed internationalization has a standard format; you can see how it looks and how it should be implemented with the stats command.

File: src/commands/Miscellaneous/discordstatus.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+					.setDescription(msg.language.tget('COMMAND_DISCORD_STAUTS_EMBED_DECRIPTION', json.status.indicator))
+					.addFields(json.components.map((component: { name: any; status: any }) => ({
+						name: component.name,
+						value: component.status,
+						inline: true
+					})))
+					.setFooter(msg.language.tget('COMMAND_DISCORD_STATUS_EMBED_FOOTER', formatDate(json.page.updated_at)));
+
+				return msg.channel.send(embed);
+			});
```
Comment (jwford) [inline src/commands/Miscellaneous/discordstatus.ts:40]: I think some error handling would be good, you know how steve likes to be temperamental sometimes 🤣

File: src/lib/types/Languages.d.ts [hunk 3]
```diff
@@ -138,6 +138,9 @@ declare module 'klasa' {
 			FOOTER: string;
 			TITLE: string;
 		};
+		COMMAND_DISCORD_STAUTS_DECRIPTION: string;
+		COMMAND_DISCORD_STAUTS_EMBED_DECRIPTION: (incident: string) => string;
+		COMMAND_DISCORD_STATUS_EMBED_FOOTER: (time: string) => string;
```
Comment (jwford) [inline src/lib/types/Languages.d.ts]: See above comment about standard format for embed internationalization

**Review by jwford — CHANGES_REQUESTED** [review]

File: src/languages/en-US.ts (hunk 1 above)
Comment (BenSegal855) [reply src/languages/en-US.ts]: Resolved by [24c4825](https://github.com/tuataria/steve/pull/51/commits/24c482586f52b18c33020abb67695e8921815edb)

File: src/commands/Miscellaneous/discordstatus.ts (hunk 2 above)
Comment (BenSegal855) [reply src/commands/Miscellaneous/discordstatus.ts:40]: Resolved by [5315f20](https://github.com/tuataria/steve/pull/51/commits/5315f208685c44e18ef5a5520a971164ee934045)

File: src/lib/types/Languages.d.ts (hunk 3 above)
Comment (BenSegal855) [reply src/lib/types/Languages.d.ts]: Resolved by [24c4825](https://github.com/tuataria/steve/pull/51/commits/24c482586f52b18c33020abb67695e8921815edb)

File: src/commands/Miscellaneous/discordstatus.ts [hunk 4]
```diff
@@ -17,21 +17,26 @@ export default class extends SteveCommand {
 
 	public async run(msg: KlasaMessage): Promise<Message> {
 		const url = 'https://srhpyqt94yxb.statuspage.io/api/v2/summary.json';
+		const EMBED_DATA = msg.language.tget('COMMAND_DISCORD_STAUTS_EMBED');
```
Comment (jwford) [inline src/commands/Miscellaneous/discordstatus.ts:20]: typo ("STAUTS")

File: src/commands/Miscellaneous/discordstatus.ts [hunk 5]
```diff
@@ -17,21 +17,26 @@ export default class extends SteveCommand {
...
 						inline: true
 					})))
-					.setFooter(msg.language.tget('COMMAND_DISCORD_STATUS_EMBED_FOOTER', formatDate(json.page.updated_at)));
+					.setFooter(EMBED_DATA.FOOTER(formatDate(json.page.updated_at)));
 
 				return msg.channel.send(embed);
+			})
+			.catch(err => {
+				console.log(err);
+				return msg.sendLocale('COMMAND_DSICORD_STATUS_ERROR');
```
Comment (jwford) [inline src/commands/Miscellaneous/discordstatus.ts]: typo ("DSICORD")

File: src/commands/Miscellaneous/discordstatus.ts [hunk 6]
```diff
@@ -17,21 +17,26 @@ export default class extends SteveCommand {
...
 		const url = 'https://srhpyqt94yxb.statuspage.io/api/v2/summary.json';
+		const EMBED_DATA = msg.language.tget('COMMAND_DISCORD_STAUTS_EMBED');
 
 		return fetch(url, { method: 'Get' })
 			.then(res => res.json())
 			.then(json => {
 				const embed = new MessageEmbed()
 					.setTitle(json.status.description)
-					.setDescription(msg.language.tget('COMMAND_DISCORD_STAUTS_EMBED_DECRIPTION', json.status.indicator))
+					.setDescription(EMBED_DATA.DECRIPTION(json.status.indicator))
```
Comment (jwford) [inline src/commands/Miscellaneous/discordstatus.ts:27]: typo ("DECRIPTION")

**Review by jwford — CHANGES_REQUESTED** [review]
this is almost ready to merge, just some typos in the language keys

**Comment by BenSegal855** [comment]
Spelling issues resolved by [c1fc6a7](https://github.com/tuataria/steve/pull/51/commits/c1fc6a7677c72f09a812de4f3e567690c135a253)

**Review by jwford — APPROVED** [review]

**Pull request merged**